nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -183,6 +183,15 @@ in
 
   config = mkIf enableDHCP {
 
+    assertions = [ {
+      # dhcpcd doesn't start properly with malloc ∉ [ libc scudo ]
+      # see https://github.com/NixOS/nixpkgs/issues/151696
+      assertion =
+        dhcpcd.enablePrivSep
+          -> elem config.environment.memoryAllocator.provider [ "libc" "scudo" ];
+      message = "dhcpcd with privilege separation is incompatible with chosen system malloc";
```

Comment:
How about specifying which ones are valid?

Reply:
perhaps:

> services.dhcpcd.enablePrivSep can only be true if environment.memoryAlloctaor.provider is either libc or scudo. <anotherone> is not compatible.

Reply:
unfortunately `enablePrivSep` is a package setting rather than a nixos option, so not sure how to reference it in a help text. maybe something extra along the lines of "you can override dhcpcd with enablePrivSep=false to use other allocators"?